wagtail_newsletter/templates/wagtail_newsletter/panels/newsletter_panel.html

```diff
@@ -51,118 +47,167 @@
 
     {% else %}
 
-        {% if has_action_permission %}
-            <div class="help-block help-info">
-                {% icon name="help" %}
-                These actions will save a new page revision.
+        <p>
+            <div class="wn-panel--buttons">
+                {% if has_action_permission.save_campaign %}
+                    <button
+                        type="button"
+                        class="button button-secondary button-longrunning"
+                        data-a11y-dialog-show="wn-save-dialog"
+                        data-controller="w-progress"
+                        data-action="wn-submit:save_campaign@window->w-progress#activate"
+                    >
+                        {% icon name="spinner" %}
+                        Save campaign to {{ backend_name }}
+                    </button>
+                {% endif %}
+
+                {% if has_action_permission.send_test_email %}
+                    <button
+                        type="button"
+                        class="button button-secondary button-longrunning"
+                        data-a11y-dialog-show="wn-test-dialog"
+                        data-controller="w-progress"
+                        data-action="wn-submit:send_test_email@window->w-progress#activate"
+                    >
+                        {% icon name="spinner" %}
+                        Send test email
+                    </button>
+                {% endif %}
+
+                {% if has_action_permission.send_campaign %}
+                    <button
+                        type="button"
+                        class="button button-primary button-longrunning"
+                        data-controller="w-progress"
+                        data-wn-panel-target="sendButton"
+                        data-action="
+                            wn-panel#clickSend
+                            wn-submit:send_campaign@window->w-progress#activate
+                        "
+                    >
+                        {% icon name="spinner" %}
+                        Send campaign
+                    </button>
+                {% endif %}
             </div>
-        {% endif %}
 
-        <p>
-            <input
-                type="hidden"
-                name="newsletter-test-email"
-                data-wn-panel-target="testAddress"
-            >
+            {% dialog icon_name="mail" id="wn-save-dialog" title="Save campaign" dialog_root_selector="[data-edit-form]" %}
```

Comment:
If the subtitle is always the same, we can extract it to a variable first, either using `{% trans %}` (with `noop` as we're not doing translations yet) or `{% fragment %}` e.g.

```html+django
{% load i18n %}
{% trans "This action will save a draft page revision." noop as dialog_subtitle %}
```

or

```html+django
{% fragment as dialog_subtitle %}This action will save a draft page revision.{% endfragment %}
```

and then

```suggestion
            {% dialog icon_name="mail" id="wn-save-dialog" title="Save campaign" dialog_root_selector="[data-edit-form]" subtitle=dialog_subtitle %}
```

(and pass `subtitle=dialog_subtitle` for the other two dialogs as well.)


Though to be honest I'm not sure if "This action will save a draft page revision." is clear. To me, my immediate thought was "why do I need to know that?"

Maybe it could be contextualized for each dialog, e.g.

- `This action will save a new draft page revision and save the newsletter campaign to {{ backend_name }} with your changes.`
- `This action will save a new draft page revision and send a test email with your changes.`
- `This action will save a new draft page revision and send the campaign using {{ backend_name }}.`

Reply:
Those messages are so much better, thank you!